### PR TITLE
feat(nuke): extended nuke protection

### PR DESCRIPTION
**Pull request by almostSouji**
- [x] In hidden operation: only allow dry-run
- [x] Reject if no members are filtered
- [x] Let dry-run execute without resolving the command
- [x] Prohibit concurrent use of anti raid nuke in the same server 
- [x] Integrate dry-run into preview

> **Note**
> Requires command deploy

### CODE REVIEW

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 1]
```diff
@@ -130,6 +139,12 @@ export async function handleAntiRaidNuke(
 	const potentialHitsDate = dayjs().format(DATE_FORMAT_LOGFILE);
 	const { creationRange, joinRange } = formatMemberTimestamps(members);
 
+	const row = createMessageActionRow([cancelButton]);
+	if (!hide) {
+		row.components.push(banButton);
+	}
+	row.components.push(dryRunButton);
```
Comment (iCrawl) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:146]:
```suggestion
	let buttons =[cancelButton];
	if (!hide) {
		buttons =  [...buttons, banButton];
	}
	const row = createMessageActionRow([...buttons, dryRunButton]);
```

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 2]
```diff
@@ -32,51 +34,135 @@ import { logger } from '../../../../logger.js';
...
+	const start = performance.now();
+	await collectedInteraction.deferUpdate();
 
-	for (const id of ids) {
-		const member = fetchedMembers.get(id);
+	const { result, cases } = await blastOff(
+		collectedInteraction,
+		{
+			days: pruneDays,
+			dryRun: false,
```
Comment (JPBM135) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:60]: Why the dry run is set to false here? I assume debugging?

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 3]
```diff
@@ -32,51 +34,135 @@ import { logger } from '../../../../logger.js';
...
-		validMembers: result,
-		validIdCount: result.size,
-		invalidIdCount: ids.size - result.size,
-		totalIdCount: ids.size,
+	const membersHitDate = dayjs().format(DATE_FORMAT_LOGFILE);
+	const formatterArgs = {
+		...args,
+		insensitive,
+		days: pruneDays as ArgsParam<typeof AntiRaidNukeCommand>['filter']['days'],
+		dryRun: false,
```
Comment (JPBM135) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:89]: Here too

**Comment by JPBM135** [comment]
Sorry, I'm dumb

File: packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts [hunk 4]
```diff
@@ -143,114 +241,51 @@ export async function handleAntiRaidNuke(
...
+					break;
+				}
+
+				case banKey: {
+					await launchNuke(collectedInteraction, members, pruneDays, insensitive, mode, args, locale);
+					buttonCollector.stop('nuke');
+					break;
+				}
+				case dryRunKey: {
+					await sendDryRunResult(collectedInteraction, members, pruneDays, insensitive, mode, args, locale);
```
Comment (JPBM135) [inline packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:272]: Why the dry run does not stop the collector?

Comment (JPBM135) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:272]: I actually think that it should stop the collector if ephemeral

Comment (almostSouji) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:272]: More and more leaning towards running dry-run before even presenting ban and cancel, and just not presenting the buttons at all, if hidden, effectively replacing the preview with a dry-run

Comment (JPBM135) [reply packages/yuudachi/src/commands/moderation/sub/anti-raid-nuke/coreCommand.ts:272]: I actually think this is a big brain ideia

File: packages/yuudachi/src/functions/logging/upsertAntiRaidArchiveLog.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import dayjs from 'dayjs';
-import { ButtonStyle, type Message, type Guild, type User } from 'discord.js';
+import { ButtonStyle, type Message, type Guild, type User, ButtonInteraction } from 'discord.js';
```
Comment (iCrawl) [inline packages/yuudachi/src/functions/logging/upsertAntiRaidArchiveLog.ts:2]:
```suggestion
import { ButtonStyle, type Message, type Guild, type User, type ButtonInteraction } from 'discord.js';
```

**Pull request merged**